src/class-wpml-elementor-translate-ids.php

```diff
@@ -103,6 +104,25 @@ public function translate_global_widget_ids( $data_array, $post_id ) {
 		return $data_array;
 	}
 
+	/**
+	 * @param array $data_array
+	 * @param int   $post_id
+	 *
+	 * @return array
+	 */
+	public function translate_product_ids( $data_array, $post_id ) {
+		foreach ( $data_array as &$data ) {
+			if ( isset( $data['elType'] ) && 'widget' === $data['elType'] ) {
```

Comment:
You can simplify with only 1 `if`:

```php
if ( Obj::prop( 'elType', $data ) === 'widget' && Obj::prop( 'widgetType', $data ) === 'wc-add-to-cart' ) {
    $data['settings']['product_id'] = $this->translate_id( $data['settings']['product_id'] );
}
```

If yo want to go the extra mile, this is a common pattern to use functional programming with `over` and a `lensProp`.

Now that you have the unit tests you could even try to refactor your code with a full FP pipeline.

Reply:
I am giving up @strategio 

I need to be more familiar with the different patterns to be able to apply them.

I did apply your suggestion though